dms/models/dms_file.py

```diff
@@ -133,6 +133,27 @@ class File(models.Model):
         attachment=True, string="Content File", prefetch=False, invisible=True
     )
 
+    @api.depends("res_mimetype")
+    def _compute_custom_thumbnail_editable(self):
+        """When a dms.file is an image, forbid editing thumbnail."""
+        result = super()._compute_custom_thumbnail_editable()
+        for one in self.filtered("res_mimetype"):
+            if one.res_mimetype.startswith("image/"):
+                one.custom_thumbnail_editable = False
+        return result
+
+    @api.depends(
+        "custom_thumbnail_medium", "custom_thumbnail_small", "res_mimetype", "content"
+    )
+    def _compute_thumbnail(self):
+        """Provide image thumbnail automatically if possible."""
+        result = super()._compute_thumbnail()
+        for one in self.filtered("res_mimetype"):
+            if one.res_mimetype.startswith("image/"):
+                for field in ("thumbnail", "thumbnail_medium", "thumbnail_small"):
+                    one[field] = one.content
```

Comment:
This strategy is not adequate, as it will provide the full image to all thumbnail sizes. You should use image resize methods for providing the proper one. I'm also worried about the performance of this not being stored. What about putting the image as custom thumbnail on content changed and let the rest to work as is?

Reply:
This was implemented in #88 on purpose, for performance. When loading multiple records, it turns out to be more heavy having to resize twice each image than just setting all thumbnails the same size and letting the browser adjust them. Keep in mind that this is computed and non-stored, so it'd spend lots of CPU cycles resizing once and again the same images.

In reality, most chances are today we could just remove all those 3 fields, and rely on using a custom thumbnail if available, or the image URL otherwise. But these were kept to ease the instability caused by so many patches on standard versions.

One alternative would be to split this into 3 methods, so the ORM only lazy-computes the ones really needed, but that'd turn out to have them never used for the reason just explained, so this seemed like the simplest solution.

I can add the removal of those fields if you think it's worth it at this point.

Reply:
One of the purposes of extra fields is to avoid sending a lot of big files through Internet for only previewing something. This is what `image.mixin` does. And for me, only 2 main fields should be exists: icon and custom_image/custom_preview, but with the automatic extension to lower resolutions provided by `image.mixin`. Can you do this switch?

Reply:
Let me check.